src/Mooc/Courses/Application/Create/CourseCreator.php

```diff
@@ -22,11 +24,18 @@ public function __construct(CourseRepository $repository, EventBus $bus)
         $this->bus        = $bus;
     }
 
-    public function __invoke(CourseId $id, CourseName $name, CourseDuration $duration): void
+    public function __invoke(CourseId $id, CourseName $name, CourseDuration $duration)
     {
         $course = Course::create($id, $name, $duration);
+        $courseFound = $this->repository->search($id);
+        if(null === $courseFound)
+        {
+            $this->repository->save($course);
+            $this->bus->publish(...$course->pullDomainEvents());
+            
+        }else{
+            throw new CourseDuplicated($id);
+        }
```

Comment:
```suggestion
        $this->ensureCourseExist();
      
        $this->repository->save($course);
        $this->bus->publish(...$course->pullDomainEvents());
```

Extract to a guard clause like that